sentrysql/stmt.go

```diff
@@ -0,0 +1,190 @@
+package sentrysql
+
+import (
+	"context"
+	"database/sql/driver"
+
+	"github.com/getsentry/sentry-go"
+)
+
+type sentryStmt struct {
+	originalStmt driver.Stmt
+	query        string
+	ctx          context.Context
+	config       *sentrySQLConfig
+}
+
+func (s *sentryStmt) Close() error {
+	return s.originalStmt.Close()
+}
+
+func (s *sentryStmt) NumInput() int {
+	return s.originalStmt.NumInput()
+}
+
+//nolint:dupl
+func (s *sentryStmt) Exec(args []driver.Value) (driver.Result, error) {
+	parentSpan := sentry.SpanFromContext(s.ctx)
+	if parentSpan == nil {
+		return s.originalStmt.Exec(args) //nolint:staticcheck We must support legacy clients
+	}
+
+	span := parentSpan.StartChild("db.sql.exec", sentry.WithDescription(s.query))
+	if s.config.databaseSystem != "" {
+		span.SetData("db.system", s.config.databaseSystem)
+	}
+	if s.config.databaseName != "" {
+		span.SetData("db.name", s.config.databaseName)
+	}
+	if s.config.serverAddress != "" {
+		span.SetData("server.address", s.config.serverAddress)
+	}
+	if s.config.serverPort != "" {
+		span.SetData("server.port", s.config.serverPort)
+	}
+
+	result, err := s.originalStmt.Exec(args) //nolint:staticcheck We must support legacy clients
+	if err != nil {
+		span.Status = sentry.SpanStatusInternalError
+		span.Finish()
+		return nil, err
+	}
+
+	span.Status = sentry.SpanStatusOK
+	span.Finish()
+
+	return result, nil
+}
+
+//nolint:dupl
+func (s *sentryStmt) Query(args []driver.Value) (driver.Rows, error) {
+	parentSpan := sentry.SpanFromContext(s.ctx)
+	if parentSpan == nil {
+		return s.originalStmt.Query(args) //nolint:staticcheck We must support legacy clients
+	}
+
+	span := parentSpan.StartChild("db.sql.query", sentry.WithDescription(s.query))
+	if s.config.databaseSystem != "" {
+		span.SetData("db.system", s.config.databaseSystem)
+	}
+	if s.config.databaseName != "" {
+		span.SetData("db.name", s.config.databaseName)
+	}
+	if s.config.serverAddress != "" {
+		span.SetData("server.address", s.config.serverAddress)
+	}
+	if s.config.serverPort != "" {
+		span.SetData("server.port", s.config.serverPort)
+	}
+
+	rows, err := s.originalStmt.Query(args) //nolint:staticcheck We must support legacy clients
+	if err != nil {
+		span.Status = sentry.SpanStatusInternalError
+		span.Finish()
+
+		return nil, err
+	}
+
+	span.Status = sentry.SpanStatusOK
+	span.Finish()
+
+	return rows, nil
+}
+
+func (s *sentryStmt) ExecContext(ctx context.Context, args []driver.NamedValue) (driver.Result, error) {
+	// should only be executed if the original driver implements StmtExecContext
+	stmtExecContext, ok := s.originalStmt.(driver.StmtExecContext)
+	if !ok {
+		// fallback to the so-called deprecated "Exec" method
+		var values []driver.Value
```

Comment:
You can preallocate this.

Reply:
I'm gonna need to read the driver package docs again, see if I can just throw driver.ErrSkipped (or ErrSkip, I forgot)